gobblin-cluster/src/main/java/org/apache/gobblin/cluster/ClusterEventMetadataGenerator.java

```diff
@@ -43,17 +43,21 @@
     List<TaskState> taskStates = jobContext.getJobState().getTaskStates();
     String taskException = EventMetadataUtils.getTaskFailureExceptions(taskStates);
     String jobException = EventMetadataUtils.getJobFailureExceptions(jobContext.getJobState());
-
+    Map<String,String> jobMetadata = new HashMap<>();
+    jobMetadata.put(TimingEvent.FlowEventConstants.HIGH_WATERMARK_FIELD, jobContext.getJobState().getProp(TimingEvent.FlowEventConstants.HIGH_WATERMARK_FIELD, ""));
+    jobMetadata.put(TimingEvent.FlowEventConstants.LOW_WATERMARK_FIELD, jobContext.getJobState().getProp(TimingEvent.FlowEventConstants.LOW_WATERMARK_FIELD, ""));
     switch (eventName) {
       case JOB_COMPLETE:
-        return ImmutableMap.of(PROCESSED_COUNT_KEY, Long.toString(EventMetadataUtils.getProcessedCount(taskStates)));
+        jobMetadata.put(PROCESSED_COUNT_KEY, Long.toString(EventMetadataUtils.getProcessedCount(taskStates)));
```

Comment:
You can use ImmutableMap.Builder and keep the returned object Immutable

Reply:
addressed your comments